### PR TITLE
Feat: added likes/dislike feature

### PR DESCRIPTION
**Pull request by poornima-alugubelly**
User can like and undo like from explore page and video page , the updates can be seen in liked playlist

### CODE REVIEW

**Comment by horsemaker** [comment]
image

I am not sure if you are aware of this but this avatar ain't visible.

File: src/pages/Explore/components/VideoCard.jsx [hunk 1]
```diff
@@ -2,26 +2,76 @@ import "./VideoCard.css";
...
+				{openOptions && (
+					<ul className="video-option-container">
+						<li class="list-item flex-row gap-xs" onClick={() => likeHandler()}>
+							{inLikedVideos ? (
+								<i class="far fa-check-circle"></i>
+							) : (
+								<i class="fas fa-plus"></i>
+							)}
+							Liked Videos
+						</li>
```
Comment (kshitijthatte) [inline src/pages/Explore/components/VideoCard.jsx]: You can do a check here if the user is not logged in either you can navigate them to the login page or maybe give an alert to inform them.

**Comment by poornima-alugubelly** [comment]
> image
> 
> I am not sure if you are aware of this but this avatar ain't visible.

yes yes ik

**Comment by horsemaker** [comment]
Also, when the user the logged out the liked videos are still visible. They shouldn't be, right?

image

File: src/context/UserDataContext.js [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+							type: SET_WATCHLATER,
+							payload: { watchLaterPlaylist: res.data.watchLater },
+						});
+						setLoading(false);
+					}
+				} catch (err) {
+					console.log("error", err);
+				}
+			})();
+	}, [auth.isAuthVL]);
```
Comment (horsemaker) [inline src/context/UserDataContext.js:62]: You could maybe do a cleanup here for liked videos & watch later when the use logs out.

File: src/pages/Explore/components/VideoCard.jsx [hunk 3]
```diff
@@ -2,26 +2,76 @@ import "./VideoCard.css";
...
+						</div>
+					</div>
+					<i
+						class="fas fa-ellipsis-v"
+						onClick={() => setOpenOptions(!openOptions)}
+					></i>
 				</div>
-				<i class="fas fa-ellipsis-v" onClick={() => setOpened(true)}></i>
+				{openOptions && (
+					<ul className="video-option-container">
```
Comment (kshitijthatte) [inline src/pages/Explore/components/VideoCard.jsx:56]: I really like this way of giving multiple options here.

File: src/hooks/useRemoveFromPlaylist.js [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+					type: SET_LIKES,
+					payload: { likesPlaylist: res.data.likes },
+				});
+			}
+		} catch (err) {
+			toast.error(...err.response.data.errors);
+		}
+	};
+	return [removeFromPlaylistServerCall, updatingPlaylist, setUpdatingPlaylist];
+};
```
Comment (horsemaker) [inline src/hooks/useRemoveFromPlaylist.js:28]: Interesting Custom Hooks! 👌

**Review by manojasarsa — APPROVED** [review]
On clicking on like btn, added to playlist is been showing (on toast). It can be changed.
The UI looks super cool 💯 
Good to merge!!

**Review by horsemaker — APPROVED** [review]
The code is pretty well structured! Kudos for that! 👌

**Review by kshitijthatte — APPROVED** [review]
In the likes page, on clicking the video you can redirect the user to the video page.
Great work! Good to merge.

**Pull request merged**